.upstream-tests/test/CMakeLists.txt

```diff
@@ -43,6 +43,13 @@ if (NOT MSVC)
      --compiler-options=-Wextra")
 endif()
 
+# Intel OneAPI compiler has fast math enabled by default which breaks almost all floating point tests
+if (${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM")
+  set(LIBCUDACXX_TEST_COMPILER_FLAGS
+    "${LIBCUDACXX_TEST_COMPILER_FLAGS} \
+    --compiler-options=-fno-fast-math")
+endif()
```

Comment:
These options seem to be targeting nvcc (nvc++?) when icpx is the host compiler. Should this code be under `if (${CMAKE_CUDA_COMPILER_ID} STREQUAL "NVIDIA")`?

Reply:
AFAIK this should be applied always because of our libcxx tests where we pretent to be a standard library

Reply:
I specifically disabled libcxx tests with this compiler. We can turn them on later, but I wanted our most important coverage to come first.